### PR TITLE
Add terraform workspace to prompt

### PR DESCRIPTION
**Pull request by ofirshtrull**

### CODE REVIEW

File: themes/powerline/powerline.base.bash [hunk 1]
```diff
@@ -41,6 +41,15 @@ function __powerline_user_info_prompt {
...
 
+function __powerline_terraform_prompt() {
+  local terraform_workspace=""
+
+  if [ -d .terraform ]; then
+    terraform_workspace="$(command terraform workspace show 2>/dev/null)"
+    [[ -n "${terraform_workspace}" ]] && echo "${TERRAFORM_CHAR}${terraform_workspace}|${TERRAFORM_THEME_PROMPT_COLOR}"
+  fi
+}
+
```
Comment (davidpfarrell) [inline themes/powerline/powerline.base.bash]: The core detection logic here could be placed in `~/themes/base.theme.bash` which would then be usable by all prompts in the system.  Perhaps something like (untested) :

_base.theme.bash_
```
function terraform_workspace_prompt {
    if _command_exists terraform ; then
        if [ -d .terraform ]; then
            echo -e "$(terraform workspace show 2>/dev/null)"
        fi
    fi
}
```

_powerline.base.bash_
```
function __powerline_terraform_prompt() {
    local terraform_workspace=""

    terraform_workspace="$(terraform_workspace_prompt)"
    [[ -n "${terraform_workspace}" ]] && echo "${TERRAFORM_CHAR}${terraform_workspace}|${TERRAFORM_THEME_PROMPT_COLOR}"
  fi
}
```

File: themes/powerline-naked/powerline-naked.theme.bash [hunk 2]
```diff
@@ -47,6 +47,9 @@ RUBY_CHAR=${POWERLINE_RUBY_CHAR:="❲r❳ "}
 KUBERNETES_CONTEXT_THEME_CHAR=${POWERLINE_KUBERNETES_CONTEXT_CHAR:="⎈ "}
 KUBERNETES_CONTEXT_THEME_PROMPT_COLOR=${POWERLINE_KUBERNETES_CONTEXT_COLOR:=26}
 
+TERRAFORM_THEME_PROMPT_COLOR=${POWERLINE_TERRAFORM_COLOR:=161}
+TERRAFORM_CHAR=${POWERLINE_TERRAFORM_CHAR:="❲t❳ "}
+
```
[the remark this answers is not in the document]
Comment (ofirshtrull) [reply themes/powerline-naked/powerline-naked.theme.bash]: Yes I have a black background and got the desired result 
`terraform/main > ofir > dev > `

File: themes/powerline/README.md [hunk 3]
```diff
@@ -67,6 +67,7 @@ The contents of the prompt can be "reordered", all the "segments" (every piece o
 * `node` - Current node version (only `nvm` is supported)
 * `scm` - Version control information, `git`
 * `user_info` - Current user
+* `terraform` - Current terraform workspace
```
[the remark this answers is not in the document]
Comment (ofirshtrull) [reply themes/powerline/README.md]: done

File: themes/powerline-plain/README.md [hunk 4]
```diff
@@ -52,6 +52,7 @@ The contents of the prompt can be "reordered", all the "segments" (every piece o
 * `hostname` - Host name of machine
 * `in_vim` - Show identifier if running in `:terminal` from vim
 * `k8s_context` - Show current kubernetes context
+* `terraform` - Current terraform workspace
```
[the remark this answers is not in the document]
Comment (ofirshtrull) [reply themes/powerline-plain/README.md]: done

File: themes/powerline-multiline/README.md [hunk 5]
```diff
@@ -66,6 +66,8 @@ The contents of both prompt sides can be "reordered", all the "segments" (every
 * `ruby` - Current ruby version if using `rvm`
 * `node` - Current node version (only `nvm` is supported)
 * `scm` - Version control information, `git` 
+* `terraform` - Current terraform workspace
+* `k8s_context` - Show current kubernetes context
```
[the remark this answers is not in the document]
Comment (ofirshtrull) [reply themes/powerline-multiline/README.md]: done

File: themes/powerline/powerline.base.bash (hunk 1 above)
Comment (ofirshtrull) [reply themes/powerline/powerline.base.bash]: done

File: themes/powerline-naked/powerline-naked.base.bash [hunk 6]
```diff
@@ -25,9 +26,9 @@ function __powerline_left_segment {
 
   LEFT_PROMPT+="$(set_color ${params[1]} -)${pad_before_segment}${params[0]}${normal}"
   LAST_SEGMENT_COLOR=${params[1]}
-  (( SEGMENTS_AT_LEFT += 1 ))
+  ((SEGMENTS_AT_LEFT += 1))
 }
 
-function __powerline_left_last_segment_padding {
+function __powerline_left_last_segment_padding() {
```
Comment (davidpfarrell) [inline themes/powerline-naked/powerline-naked.base.bash]: You can revert all the changes to this file.   We don't want to go down a rabbit hole of mixing formatting changes with new features.

File: themes/powerline/powerline.base.bash [hunk 7]
```diff
@@ -41,6 +41,24 @@ function __powerline_user_info_prompt {
...
```
Comment (davidpfarrell) [inline themes/powerline/powerline.base.bash]: Looks like you forgot to delete the older version of the function from the PR?

File: themes/powerline/powerline.base.bash [hunk 8]
```diff
@@ -41,6 +41,24 @@ function __powerline_user_info_prompt {
...
+function __powerline_terraform_prompt() {
+  local terraform_workspace=""
+
+  if [ -d .terraform ]; then
+    terraform_workspace="$(terraform_workspace_prompt)"
+    [[ -n "${terraform_workspace}" ]] && echo "${TERRAFORM_CHAR}${terraform_workspace}|${TERRAFORM_THEME_PROMPT_COLOR}"
+  fi
+}
+
```
Comment (davidpfarrell) [inline themes/powerline/powerline.base.bash]: * I didn't notice the `__powerline_terraform_prompt()` at first, but the general pattern for functions here is to leave out the `()`.

* Technically you don't need the `[ -d .terraform ]` check, but shouldn't hurt to leave it.

File: themes/powerline-naked/powerline-naked.base.bash (hunk 6 above)
Comment (ofirshtrull) [reply themes/powerline-naked/powerline-naked.base.bash]: sublime automatically adds this for me :-)
I have removed it

File: themes/powerline/powerline.base.bash (hunk 7 above)
Comment (ofirshtrull) [reply themes/powerline/powerline.base.bash]: Fixed

File: themes/powerline/powerline.base.bash (hunk 8 above)
Comment (ofirshtrull) [reply themes/powerline/powerline.base.bash]: sublime automatically adds this for me :-)
I have removed it

and I think it is good practice to leave the check

File: themes/powerline-naked/powerline-naked.base.bash [hunk 9]
```diff
@@ -25,7 +26,7 @@ function __powerline_left_segment {
 
   LEFT_PROMPT+="$(set_color ${params[1]} -)${pad_before_segment}${params[0]}${normal}"
   LAST_SEGMENT_COLOR=${params[1]}
-  (( SEGMENTS_AT_LEFT += 1 ))
+  ((SEGMENTS_AT_LEFT += 1))
```
Comment (davidpfarrell) [inline themes/powerline-naked/powerline-naked.base.bash]: Sorry my note was regarding reverting ALL changes in this file as they don't relate to the PR.

File: themes/powerline-naked/powerline-naked.base.bash [hunk 10]
```diff
@@ -1,7 +1,8 @@
 . "$BASH_IT/themes/powerline/powerline.base.bash"
 
-function __powerline_left_segment {
-  local OLD_IFS="${IFS}"; IFS="|"
+function __powerline_left_segment() {
+  local OLD_IFS="${IFS}"
+  IFS="|"
```
Comment (davidpfarrell) [inline themes/powerline-naked/powerline-naked.base.bash]: Sorry my note was regarding reverting ALL changes in this file as they don't relate to the PR.

File: themes/powerline-naked/powerline-naked.base.bash (hunk 6 above)
Comment (davidpfarrell) [reply themes/powerline-naked/powerline-naked.base.bash]: Ahh the curse of auto-formatting :)

I forgot to leave a note on the other changes in the file, but the same applies there.

File: themes/powerline-naked/powerline-naked.base.bash (hunk 10 above)
Comment (ofirshtrull) [reply themes/powerline-naked/powerline-naked.base.bash]: fixed

File: themes/powerline-naked/powerline-naked.base.bash (hunk 9 above)
Comment (ofirshtrull) [reply themes/powerline-naked/powerline-naked.base.bash]: fixed

File: themes/powerline-multiline/powerline-multiline.theme.bash [hunk 11]
```diff
@@ -49,6 +49,9 @@ RBENV_THEME_PROMPT_SUFFIX=""
 RUBY_THEME_PROMPT_COLOR=${POWERLINE_RUBY_COLOR:=161}
 RUBY_CHAR=${POWERLINE_RUBY_CHAR:="❲r❳ "}
 
+TERRAFORM_THEME_PROMPT_COLOR=${POWERLINE_TERRAFORM_COLOR:=161}
+TERRAFORM_CHAR=${POWERLINE_TERRAFORM_CHAR:="❲t❳ "}
+
```
Comment (davidpfarrell) [inline themes/powerline-multiline/powerline-multiline.theme.bash]: For consistency, we should place new configurations in the same place in each of the powerline theme files.

This way, you can do a `diff` of each file and see only  the meaningful differences between the files.

In this file, its your new configurations are above Kubernetes, but in other files its below Kubernetes.

I don't have a preference either way, as longs its consistent in all files.

File: themes/powerline-naked/powerline-naked.theme.bash [hunk 12]
```diff
@@ -17,6 +17,9 @@ USER_INFO_SSH_CHAR=${POWERLINE_USER_INFO_SSH_CHAR:=" "}
 USER_INFO_THEME_PROMPT_COLOR=${POWERLINE_USER_INFO_COLOR:=240}
 USER_INFO_THEME_PROMPT_COLOR_SUDO=${POWERLINE_USER_INFO_COLOR_SUDO:=202}
 
+TERRAFORM_THEME_PROMPT_COLOR=${POWERLINE_TERRAFORM_COLOR:=161}
+TERRAFORM_CHAR=${POWERLINE_TERRAFORM_CHAR:="❲t❳ "}
+
```
Comment (davidpfarrell) [inline themes/powerline-naked/powerline-naked.theme.bash]: Looks like this somehow got duplicated in this file?

File: themes/powerline-multiline/powerline-multiline.theme.bash (hunk 11 above)
Comment (ofirshtrull) [reply themes/powerline-multiline/powerline-multiline.theme.bash]: good point, I have fixed the order

File: themes/powerline-naked/powerline-naked.theme.bash (hunk 12 above)
Comment (ofirshtrull) [reply themes/powerline-naked/powerline-naked.theme.bash]: removed duplicated code

**Comment by davidpfarrell** [comment]
OK my work here is done - It took a bit of back-and-forth but I think its ready to go.

Hopefully those with merge-powers will be pleased with the work we've done here :)

**Comment by ofirshtrull** [comment]
Thanks for your help, you gave me good advice for future code

**Comment by nwinkler** [comment]
Looks great - thanks for the hard work on this!

**Pull request merged**